test/Shared/UserManagerTestBase.cs

```diff
@@ -82,15 +83,16 @@ protected RoleManager<TRole> CreateRoleManager(object context = null, IServiceCo
         protected abstract void AddUserStore(IServiceCollection services, object context = null);
         protected abstract void AddRoleStore(IServiceCollection services, object context = null);
 
-        protected TUser CreateTestUser(string namePrefix = "")
-        {
-            return new TUser() { UserName = namePrefix + Guid.NewGuid().ToString() };
-        }
+        protected abstract void SetUserPasswordHash(TUser user, string hashedPassword);
 
-        protected TRole CreateRole(string namePrefix = "")
-        {
-            return new TRole() { Name = namePrefix + Guid.NewGuid().ToString() };
-        }
+        protected abstract TUser CreateTestUser(string namePrefix = "", string email = "", string phoneNumber = "",
+            bool lockoutEnabled = false, DateTimeOffset? lockoutEnd = null, bool useNamePrefixAsUserName = false);
+
+        protected abstract TRole CreateTestRole(string roleName = "");
+
+        protected abstract Expression<Func<TUser, bool>> UserNamePredicate(string userName, bool contains = false);
```

Comment:
We can't simplify the tests to just do equality since we aren't cleaning up the database, so a lot of the tests use the same username prefix, like "test", and we only drop/recreate the database once per run.  Rather than trying to manually maintain user name uniqueness, I went with the approach of randomizing the names instead by always jamming on a guid, and then doing StratsWith instead of strict equality.